gitium/gitium-webhook.php

```diff
@@ -17,10 +17,45 @@
 
 header( 'Content-Type: text/html' );
 define( 'SHORTINIT', true );
-//$wordpress_loader = $_SERVER['DOCUMENT_ROOT'] . '/wp-load.php';
-$wordpress_loader = filter_input(INPUT_SERVER, 'DOCUMENT_ROOT', FILTER_SANITIZE_FULL_SPECIAL_CHARS) . '/wp-load.php';
 
-require_once $wordpress_loader;
+$current_dir = __DIR__;
+
+// Use getenv or $_SERVER to determine the document root, with sanitization and fallback
+$document_root = getenv('DOCUMENT_ROOT') 
+    ?: filter_input(INPUT_SERVER, 'DOCUMENT_ROOT', FILTER_SANITIZE_FULL_SPECIAL_CHARS) 
+    ?? $current_dir; // Use $current_dir directly as the fallback
+$wordpress_loader = rtrim($document_root, '/\\') . '/wp-load.php';
+
+if (file_exists($wordpress_loader)) {
+    require_once $wordpress_loader;
+} else {
+    // Try to locate wp-load.php dynamically
+    if (defined('WP_CONTENT_DIR')) {
+        $web_root = dirname(WP_CONTENT_DIR);
+    } else {
+        // Fallback: search upward from the current directory
+        $web_root = $current_dir;
+        while ($web_root && !file_exists($web_root . '/wp-load.php')) {
```

Comment:
This may stuck forever if `wp-load.php` is not present in the path, for example bedrock file structure will never work.

Let's try only `/../../../`. Here is an example: define an array of strings to try for `$document_root`:

```php
$try_wp_roots = [
    getenv('DOCUMENT_ROOT'),
    filter_input(INPUT_SERVER, 'DOCUMENT_ROOT', FILTER_SANITIZE_FULL_SPECIAL_CHARS) ,
    __DIR__ . "/../../../",
    ....
];

foreach($try_wp_roots as $root) {
   if ( isset($root) && file_exists($root . '/wp-load.php') ) {
       require_once ..; 
       break;
   }
}

// check wordpress included
// if ( ! defined( 'ABSPATH' ) ) { die...
```